### PR TITLE
test: re-enable test watch tests and mark as flaky

### PR DESCRIPTION
**Pull request by caspervonb**
This marks test watcher tests as flaky tests.
They still need to be re-worked as they are very flaky.

See https://github.com/denoland/deno/issues/10552

### CODE REVIEW

File: cli/tests/integration/watcher_tests.rs [hunk 1]
```diff
@@ -427,9 +427,8 @@ fn run_watch_with_import_map_and_relative_paths() {
   temp_directory.close().unwrap();
 }
 
-// TODO(bartlomieju): flaky (https://github.com/denoland/deno/issues/10552)
-#[ignore]
 #[test]
```
Comment (bartlomieju) [inline cli/tests/integration/watcher_tests.rs:430]: This should be removed

File: cli/tests/integration/watcher_tests.rs [hunk 2]
```diff
@@ -596,9 +595,8 @@ fn test_watch() {
   drop(t);
 }
 
-// TODO(bartlomieju): flaky (https://github.com/denoland/deno/issues/10552)
-#[ignore]
 #[test]
```
Comment (bartlomieju) [inline cli/tests/integration/watcher_tests.rs:598]: Ditto

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: cli/tests/integration/watcher_tests.rs (hunk 1 above)
Comment (caspervonb) [reply cli/tests/integration/watcher_tests.rs:430]: Huh, this does not match my working tree... did something weird here.

**Review by bartlomieju — APPROVED** [review]
Provisional LGTM, if it fails on `main` I will revert it

**Pull request merged**